utils/cosmos/iscn/sign.ts

```diff
@@ -37,14 +40,30 @@ export function formatISCNTxPayload(payload: ISCNRegisterPayload): ISCNSignPaylo
     for (let i = 0; i < authorNames.length; i += 1) {
       const authorName = authorNames[i]
       const authorUrl = authorUrls[i]
-      const authorId = authorWallets[i]
+      let authorId = ''
+      authorWallets[i].forEach((a: any) => {
+        if(a.type === 'cosmos'){
+          authorId =`did:cosmos:${a.address.slice(6)}`
+        }else if (a.type ==='eth' && !authorId ){
+          authorId = `did:eth:${a.address}`
+        }else{
+          return ''
+        }
+        return undefined
+      })
```

Comment:
forEach does not need a return value
A simpler way to write this with forEach
```suggestion
      authorWallets[i].forEach((a: any) => {
        if (a.type === 'cosmos') {
          authorId =`did:cosmos:${a.address.slice(6)}`
        } else if (a.type ==='eth') {
          authorId = authorId || `did:eth:${a.address}`
        }
      })
```
Using reduce() would also work